exercicios/para-casa/lojinha.py

```diff
@@ -0,0 +1,146 @@
+
+        opt = input("+ Insira a opção desejada: ")
+
+        if opt not in ["0", "1", "2", "3", "4"]:
+            print("Opção inválida, digite uma opção válida.")
+            continue
+            
+        if opt == "1":
+            adicionar_registro()
+        elif opt == "2":
+            mostrar_receita()
+        elif opt == "3":
+            mostrar_registros()
+        elif opt == "4":
+            mostrar_anos_receita()
+        elif opt == "0":
+            break
+
+# Criando a função para adicionar registro
+def adicionar_registro():
+    try:
+        ano = int(input("+ Insira o ano correspondente: "))
+        mes = input("+ Insira o mês correspondente: ")
+        faturamento = int(input("+ Insira o faturamento correspondente a esse mês: "))
+        despesas = int(input("+ Insira as despesas correspondentes a esse mês: "))
+
+        novo_registro = {
+            "ano_receita": ano,
+            "mes_receita": mes,
+            "faturamento": faturamento,
+            "despesas": despesas
+        }
+
+        dataset.append(novo_registro)
+        print("Registro adicionado com sucesso!")
+    except ValueError:
+        print("Erro: Certifique-se de inserir números inteiros para ano, faturamento e despesas.")
+
+# Criando a função mostrar receita
+def mostrar_receita():
+    mes = input("+ Insira o mês que gostaria de saber a receita: ")
+    valor_total = 0
+    receita_maior = 0
+    receita_menor = 0
+    ano_maior_receita = 0
+    ano_menor_receita = 0
+
+    for el in dataset:
+        if el["mes_receita"] == mes:
+            receita = el["faturamento"] - el["despesas"]
+            valor_total += receita
+            
+            if receita > receita_maior:
+                receita_maior = receita
+                ano_maior_receita = el["ano_receita"]
+                
+            if receita < receita_menor:
+                receita_menor = receita
+                ano_menor_receita = el["ano_receita"]
+
+    if valor_total == 0:
+        print("Nenhum registro encontrado para o mês especificado.")
+    else:
+        print(f"A soma das suas receitas é: {valor_total}.")
+        print(f"O ano {ano_maior_receita} teve a maior receita para este mês, com um valor de {receita_maior}.")
+        print(f"O ano {ano_menor_receita} teve a menor receita para este mês, com um valor de {receita_menor}.")
+
+# Criando a função para visualizar todos os registros
+def mostrar_registros():
+    if not dataset:
+        print("Não há registros disponíveis.")
+    else:
+        print("Registros:")
+        for idx, reg in enumerate(dataset, start=1):
+            print(f"Registro {idx}: {reg}")
+
+# Função para mostrar os anos com maior e menor receita
+def mostrar_anos_receita():
+    anos = set([data['ano_receita'] for data in dataset])
+    receitas_ano = {}
+    for ano in anos:
+        receitas_ano[ano] = sum(data['faturamento'] - data['despesas'] for data in dataset if data['ano_receita'] == ano)
```

Comment:
pode me explicar como chegou nessa lógica?

Reply:
Pensei em usar o for para iterar cada ano em anos, estruturando da seguinte forma:

- Em receitas_ano[ano] peguei o valor no dicionário para armazenar a receita total do ano, porque dessa forma para todo ano seria armazenado uma chave e seu valor correspondente, para que pudesse ser feito o calculo total.
- Em sum(data['faturamento'] - data['despesas'] for data in dataset if data['ano_receita'] == ano) fiz o cálculo da receita total para o ano no loop for. Dentro disso:

1. for data in dataset if data['ano_receita'] == ano: Este for seria a iteração sobre cada dicionário data no dataset, só que ele só considera os dicionários em que o valor da chave 'ano_receita' é igual ao ano atual do loop.
2. data['faturamento'] - data['despesas']: É o calculo da receita líquida subtraindo as despesas (data['despesas']) do faturamento (data['faturamento']).
3. sum(...): Usei essa função para somar todos os valores de receita líquida calculados para o ano atual do loop, para assim obter a receita total para esse ano.